### ISSUE TITLE
Libation hangs when trying to scan huge audible account (848 Titles)

### ISSUE THREAD

**Issue by CryptoJones** [rmcrackan/Libation#173]
Libation failed twice, and then hung on the third time. I have visual studio 2022 and I am currently stepping through the code to see if I can solve this myself. 

[_Log202112.log](https://github.com/rmcrackan/Libation/files/7657820/_Log202112.log)

**Comment by CryptoJones** [comment]
[parents.json.txt](https://github.com/rmcrackan/Libation/files/7658447/parents.json.txt)

**Comment by CryptoJones** [comment]
Full Library List
[books.json.txt](https://github.com/rmcrackan/Libation/files/7658545/books.json.txt)

**Comment by CryptoJones** [comment]
Found the problem. Somehow I have two books with the same ProductID in my Library

![2021-12-06 02_58_16-Libation (Debugging) - Microsoft Visual Studio](https://user-images.githubusercontent.com/1234985/144817196-9f777195-2e17-4d02-a15a-a40e31aca92e.png)

**Comment by CryptoJones** [comment]
Duplicate Key 
![2021-12-06 03_54_44-Window](https://user-images.githubusercontent.com/1234985/144825566-184d0b91-f8c7-48e9-aaf8-69849af0b06b.png)

**Comment by CryptoJones** [comment]
After setting optionsBuilder.EnableSensitiveDataLogging(); in LibationContext.c
![2021-12-06 04_13_03-Libation (Debugging) - Microsoft Visual Studio](https://user-images.githubusercontent.com/1234985/144828423-2c6c2cbf-dfcb-463e-bdf4-d874a4d3956c.png)
s

**Comment by CryptoJones** [comment]
```
		private int upsertBooks(IEnumerable<ImportItem> importItems)
		{
			var qtyNew = 0;

			foreach (var item in importItems)
			{
				bool DuplicateBook = false;
				foreach(Book book in DbContext.Books.Local)
                {
					if (book.AudibleProductId == item.DtoItem.ProductId)
                    {
						DuplicateBook = true;
                    }
                }
				if (!DuplicateBook)
                {
					var book = DbContext.Books.Local.FirstOrDefault(p => p.AudibleProductId == item.DtoItem.ProductId);
					if (book is null)
					{
						book = createNewBook(item);
						qtyNew++;
					}

					updateBook(item, book);
				}
			}

			return qtyNew;
		}
```

**Comment by CryptoJones** [comment]
{"The instance of entity type 'LibraryBook' cannot be tracked because another instance with the key value '{BookId: -2147480851}' is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached."}
[_Log202112.log](https://github.com/rmcrackan/Libation/files/7660690/_Log202112.log)

**Comment by CryptoJones** [comment]
![2021-12-06 06_35_52-robbuf - Discord](https://user-images.githubusercontent.com/1234985/144847391-d5e5c924-1bc9-4ca6-af37-6e7b616d0352.png)
[_Log202112.log](https://github.com/rmcrackan/Libation/files/7660696/_Log202112.log)

**Comment by rmcrackan** [comment]
Thanks for digging into this! I'll take a look today

**Comment by CryptoJones** [comment]
Sorry I can't be more help.  I hit the ceiling of my programming skills.

**Comment by rmcrackan** [comment]
Fixed in latest version [v6.6](https://github.com/rmcrackan/Libation/releases/latest)
